src/sage/rings/integer.pyx

```diff
@@ -4829,6 +4829,20 @@ cdef class Integer(sage.structure.element.EuclideanDomainElement):
             sage: (-64).perfect_power()                                                 # optional - sage.libs.pari
             (-4, 3)
         """
+        cdef long n
+        # Fast PARI-free path
+        if mpz_fits_slong_p(self.value):
+            n = mpz_get_si(self.value)
+            if -8 < n < 4:
+                return self, one
+            if n >= 4:
+                if not (n & 1):
+                    if mpz_popcount(self.value) == 1:
+                        return smallInteger(2), smallInteger(mpz_sizeinbase(self.value, 2) - 1)
```

Comment:
I would prefer sanity tests for this.
It's so easy to get this wrong by +-1....

Reply:
True. Added tests in d2c9c5efd2